src/go/util/request.go

```diff
@@ -18,39 +18,58 @@ import (
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"time"
 
+	"github.com/golang/glog"
 	"github.com/golang/protobuf/proto"
 )
 
-func callWithAccessToken(client *http.Client, path, method, token string) ([]byte, error) {
+var Retries = 30
+var RetryInterval = 3 * time.Second
+
+func callWithAccessToken(client *http.Client, path, method, token string) ([]byte, int, error) {
 	req, _ := http.NewRequest(method, path, nil)
 	req.Header.Add("Authorization", "Bearer "+token)
 	req.Header.Set("Content-Type", "application/x-protobuf")
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return nil, http.StatusOK, err
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("http call to %s %s returns not 200 OK: %v", method, path, resp.Status)
+		return nil, resp.StatusCode, fmt.Errorf("http call to %s %s returns not 200 OK: %v", method, path, resp.Status)
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return nil, fmt.Errorf("fail to read response body: %s", err)
+		return nil, http.StatusOK, fmt.Errorf("fail to read response body: %s", err)
 	}
 
-	return body, nil
+	return body, http.StatusOK, nil
 }
 
+// Method to call servicecontrol for latest service rolloutId and servicecontrol for service rollout and service config.
 var CallGoogleapis = func(client *http.Client, path, method string, getTokenFunc GetAccessTokenFunc, output proto.Message) error {
 	token, _, err := getTokenFunc()
 	if err != nil {
 		return fmt.Errorf("fail to get access token: %v", err)
 	}
 
-	respBytes, err := callWithAccessToken(client, path, method, token)
+	var respBytes []byte
+	var statusCode int
+
+	// Retry calls for code 429 Too Many Requests.
+	for i := 0; i < Retries; i++ {
```

Comment:
I am not sure if this code only used for calling ServieManagement,  or also for ServiceControl. 
ServiceControl should not have quota limit, so retry policy should be different.

Reply:
Yes. It is also used for servicecontrol and its quota limit is super high, so adding this won't effect calling servicecontrol.

Reply:
For new version, calling service control is not effected at all.